rero_ils/modules/documents/dojson/contrib/marc21tojson/utils.py

```diff
@@ -1735,7 +1671,237 @@ def do_work_access_point(marc21, key, value):
+
+    if fields_034 and len(fields_034) >= index:
+        """
+        * "longitude": "034$d 034$e" - concaténer les valeurs et les séparer
+          si nécessaire d'un espace
+        * "latitude": "034$f 034$g" - concaténer les valeurs et les séparer
+          si nécessaire d'un espace
+        """
+        subfields_034_d = marc21.get_subfields(fields_034[index], 'd')
+        subfields_034_e = marc21.get_subfields(fields_034[index], 'e')
+        subfields_034_f = marc21.get_subfields(fields_034[index], 'f')
+        subfields_034_g = marc21.get_subfields(fields_034[index], 'g')
+
+        longitude_parts = []
+        latitude_parts = []
+        if subfields_034_d:
+            longitude_parts.append(subfields_034_d[0])
+        if subfields_034_e:
+            longitude_parts.append(subfields_034_e[0])
+        if subfields_034_f:
+            latitude_parts.append(subfields_034_f[0])
+        if subfields_034_g:
+            latitude_parts.append(subfields_034_g[0])
+        if longitude_parts:
+            coordinates['longitude'] = ' '.join(longitude_parts)
+            coordinates['latitude'] = ' '.join(latitude_parts)
+    if coordinates:
+        cartographic_attribute['coordinates'] = coordinates
+    if cartographic_attribute:
+        cartographic_attributes.append(cartographic_attribute)
+        data['cartographicAttributes'] = cartographic_attributes
+
+
+def do_temporal_coverage(marc21, key, value):
+    """Get temporal coverage.
+
+    Plusieurs dates simples
+    Si 045 ind1=0 ou ind1=1, créer la propriété "temporalCoverage"
+    * créer un objet *date simple* de "temporalCoverage" pour chaque
+       paire 045$b et chaque 045$c
+    * "type": "time"
+    * "date": "045$b|045$c"
+    * "period_code": ["045$a"] - si répétés, récupérer tous les codes comme
+       valeurs séparées de "period_code"
+
+    Si 045 ind1=2
+    * "type": "period"
+    * "start_date": "045$b|045$c" - prendre le 1er sous-champ
+    * "end_date": "045$b|045$c" - prendre le 2e sous-champ
+    * "period_code": ["045$a"] - si répétés, récupérer tous les codes comme
+      valeurs séparées de "period_code"
+
+    Sinon (045 ind1=vide ou autre)
+    * "type": "period"
+    * "period_code": ["045$a"] - si répétés, récupérer tous les codes comme
+      valeurs séparées de "period_code"
+
+    [1] Convertir la date dans le format suivant:
+    (+/-)yyyyyyyyyyy-mm-ddTHH-MM-SS La saisie des mm, dd, HH est facultative.
+    Si HH est saisi, MM et SS doivent être saisis. Voir regex dans colonne M.
+    Les dates dans $b suivent le format (c|d)yyyymmddhh. "c" est avant JC,
+    devient -. "d" est après JC, devient +.
+    Les dates dans $c suivent le format yyyyy...
+    (autant de chiffres que nécessaires). Ces dates sont toutes négatives
+    (à convertir avec un -).
+    """
+    def test_min_max(data, minimum, maximum):
+        with contextlib.suppress(ValueError):
+            number = int(data)
+            return number >= minimum and number <= maximum
```

Comment:
`return minium <= number <= maximum`. Strange that Sourcery doens't propose this solution.